pkg/controller/daemon/daemoncontroller.go

```diff
@@ -468,6 +470,16 @@ func (dsc *DaemonSetsController) getNodesToDaemonPods(ds *extensions.DaemonSet)
 	if err != nil {
 		return nil, err
 	}
+	// After listing children, but before performing adoption, we need to recheck
+	// whether the GC has marked us as deleted (see #42639).
+	// TODO: Remove this when GC prevents races via ObservedGeneration (#26120).
```

Comment:
I think you were looking at an outdated commit. The current version of this PR does not introduce any new copies. Also based on discussion in #43032, it seems that the GC may never wait for ObservedGeneration.